crates/ruff/src/rules/pylint/rules/redefined_loop_name.rs

```diff
@@ -255,10 +254,10 @@ fn assignment_is_cast_expr(value: &Expr, target: &Expr, semantic: &SemanticModel
     semantic.match_typing_expr(func, "cast")
 }
 
-fn assignment_targets_from_expr<'a, U>(
```

Comment:
We don't need these generics, and we can remove some of the lifetimes too (unrelated to the bug fix).